Models/RepeatedMessageModel.cpp

```diff
@@ -37,9 +44,18 @@ int RepeatedMessageModel::columnCount(const QModelIndex & /*parent*/) const {
 }
 
 bool RepeatedMessageModel::setData(const QModelIndex &index, const QVariant &value, int role) {
+  R_EXPECT(index.row() >= 0 && index.row() < _subModels.size(), false) <<
+    "Supplied row was out of bounds:" << index.row();
   return _subModels[index.row()]->setData(_subModels[index.row()]->index(index.column()), value, role);
 }
 
 QVariant RepeatedMessageModel::data(const QModelIndex &index, int role) const {
+  R_EXPECT(index.row() >= 0 && index.row() < _subModels.size(), QVariant()) <<
+    "Supplied row was out of bounds:" << index.row();
+  // protobuf field number with 0 is impossible, use as sentinel to get model itself
+  if (index.column() == 0) {
+    R_EXPECT(index.isValid(), QVariant()) << "Supplied index was invalid:" << index;
```

Comment:
This one might be redundant because it just checks that the row and column are non-negative. We already check the row, because it's necessary to do so. And the column would not be negative if it was equal to 0, so I don't really think this is necessary.